src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -982,7 +983,10 @@ expectTimeout n = receiveTimeout n [match return]
 spawnAsync :: NodeId -> Closure (Process ()) -> Process SpawnRef
 spawnAsync nid proc = do
   spawnRef <- getSpawnRef
-  sendCtrlMsg (Just nid) $ Spawn proc spawnRef
+  node     <- getSelfNode
+  if nid == node
+    then sendCtrlMsg Nothing $! Spawn (DS.force proc) spawnRef
```

Comment:
I'm not sure I understood why `seq (encode x) x` is better or cheap here..
Can we have some outline here?

1. should I strictness to `sendCtrlMsg` itself (in `Nothing` branch?
2. should I use `seq (encode x) x` in order to force value in every place where we are sending (or should send but don't do it due to optimization)?
3. should I introduce changes to any other use of `sendCrtlMsg` in this PR?